src/core/locator/qgslocatorfilter.h

```diff
@@ -82,6 +82,11 @@ class CORE_EXPORT QgsLocatorResult
      */
     double score = 0.5;
 
+    /**
+      * Group the results by categores
```

Comment:
\since QGIS 3.2

I'd also like to see this doc expanded somewhat - e.g. explain that empty (default) string means all results from that filter are grouped together, but if a group string is specified than all results from that categories will be grouped together under a matching category heading. Also mention that it should be translated.